src/sinks/sematext/metrics.rs

```diff
@@ -0,0 +1,404 @@
+    }
+
+    fn call(&mut self, items: Vec<Metric>) -> Self::Future {
+        let input = encode_events(&self.config.token, items);
+        let body: Vec<u8> = input.into_bytes();
+
+        self.inner.call(body)
+    }
+}
+
+fn create_build_request(
+    uri: http::Uri,
+) -> impl Fn(Vec<u8>) -> BoxFuture<'static, crate::Result<hyper::Request<Vec<u8>>>> + Sync + Send + 'static
+{
+    move |body| {
+        Box::pin(ready(
+            hyper::Request::post(uri.clone())
+                .header("Content-Type", "text/plain")
+                .body(body)
+                .map_err(Into::into),
+        ))
+    }
+}
+
+/// Sematext takes the first part of the name as the namespace for the event.
+/// The rest of the name is the value label.
+/// If the name is not a '.' separated string, it will take the whole name as the
+/// namespace, and set label the value as 'value'.
+/// This probably should not happen if you want meaningful metrics in Sematext.
+fn split_event_name(name: &str) -> (String, String) {
+    match name.find('.') {
+        None => (name.into(), "value".into()),
+        Some(pos) => (name[0..pos].into(), name[pos + 1..].into()),
+    }
+}
+
+fn encode_events(token: &str, events: Vec<Metric>) -> String {
+    let mut output = String::new();
+    for event in events.into_iter() {
+        let fullname = event.name;
+        let (namespace, label) = split_event_name(&fullname);
+        let ts = encode_timestamp(event.timestamp);
+
+        // Authentication in Sematext is by inserting the token as a tag.
+        let mut tags = event.tags.clone().unwrap_or_else(BTreeMap::new);
+        tags.insert("token".into(), token.into());
+
+        match event.value {
+            MetricValue::Counter { value } => {
+                let fields = to_fields(label, value);
+
+                influx_line_protocol(
+                    ProtocolVersion::V1,
+                    namespace,
+                    "counter",
+                    Some(tags),
+                    Some(fields),
+                    ts,
+                    &mut output,
+                );
+            }
+            MetricValue::Gauge { value } => {
+                let fields = to_fields(label, value);
+
+                influx_line_protocol(
+                    ProtocolVersion::V1,
+                    namespace,
+                    "gauge",
+                    Some(tags),
+                    Some(fields),
+                    ts,
+                    &mut output,
+                );
+            }
+            _ => {
+                warn!(
+                    "invalid metric sent to sematext metrics sink ({:?}) ({:?})",
+                    event.kind, event.value
+                );
+            }
+        }
+    }
+
+    output.pop();
+    output
+}
+
+pub fn to_fields(label: String, value: f64) -> HashMap<String, Field> {
+    vec![(label, Field::Float(value))]
+        .into_iter()
+        .collect::<HashMap<String, Field>>()
```

Comment:
This is a lot of extra work under the hood to create a one-element `HashMap`, for no real gain.
```suggestion
    let result = HashMap::new();
    result.insert(label, value);
    result
```